pensieve/tests/integration/test_analysis_integration.py

```diff
@@ -0,0 +1,142 @@
+            job_config.autodetect = True
+
+            with open(self.events_source, "rb") as source_file:
+                job = self.client.load_table_from_file(
+                    source_file, table_ref, job_config=job_config
+                )
+
+            job.result()  # Waits for table load to complete.
+
+        # create required UDF
+        with open(self.get_key_udf, "r") as source_file:
+            self.client.query(source_file.read()).result()
+
+    def bq_execute(cls, *args, **kwargs):
+        # to use the test project and dataset, we need to change the SQL query
+        # generated by mozanalysis
+        query = args[0].replace("moz-fx-data-shared-prod", cls.project)
+        query = query.replace("telemetry", "test_data")
+
+        dataset = bigquery.dataset.DatasetReference.from_string(
+            cls.dataset, default_project=cls.project,
+        )
+
+        destination_table = None
+
+        if len(args) > 1:
+            destination_table = args[1]
+
+        kwargs = {}
+        if destination_table:
+            kwargs["destination"] = dataset.table(destination_table)
+            kwargs["write_disposition"] = bigquery.job.WriteDisposition.WRITE_TRUNCATE
+        config = bigquery.job.QueryJobConfig(default_dataset=dataset, **kwargs)
+        job = cls.client.query(query, config)
+        # block on result
+        return job.result(max_results=1)
+
+    def test_metrics(self):
+        experiment = Experiment(
+            slug="test-experiment",
+            type="rollout",
+            start_date=dt.datetime(2020, 3, 30, tzinfo=pytz.utc),
+            end_date=dt.datetime(2020, 6, 1, tzinfo=pytz.utc),
+            proposed_enrollment=7,
+            variants=[
+                Variant(is_control=False, slug="branch1", ratio=0.5),
+                Variant(is_control=True, slug="branch2", ratio=0.5),
+            ],
+            normandy_slug="test-experiment",
+        )
+
+        with mock.patch.object(BigQueryClient, "execute", new=TestAnalysisIntegration.bq_execute):
+            analysis = Analysis(self.project_id, self.test_dataset, experiment)
+
+            test_clients_daily = DataSource(
+                name="clients_daily", from_expr=f"`{self.project_id}.test_data.clients_daily`",
+            )
+
+            test_active_hours = Metric(
+                name="active_hours",
+                data_source=test_clients_daily,
+                select_expr=agg_sum("active_hours_sum"),
+            )
+
+            analysis.STANDARD_METRICS = [test_active_hours]
+
+            analysis.run(current_date=dt.datetime(2020, 4, 12), dry_run=False)
+
+        query_job = self.client.query(
+            f"""
+            SELECT
+              COUNT(*) as count
+            FROM `{self.project_id}.{self.test_dataset}.test_experiment_week_1`
+        """
+        )
+
+        result = query_job.result()
+
+        for row in result:
+            assert row["count"] == 2
```

Comment:
This is just some example test placeholder. I guess we should add some more meaningful tests as we implement more features.

Reply:
Perhaps a comment above on why we are expecting to see `row["count"] == 2`